src/react/reconcilation.js

```diff
@@ -160,29 +200,22 @@ export class Reconciler {
         if (error.name === "Invariant Violation") {
           throw error;
         }
+
+        let message;
         if (error instanceof ExpectedBailOut) {
-          this.logger.logWarning(
-            componentType,
-            `__optimizeReactComponentTree() React component tree failed due expected bail-out - ${error.message}`
-          );
-          evaluatedRootNode.message = error.message;
-          evaluatedRootNode.status = "BAIL-OUT";
+          message = `Failed to optimize React component tree for "${
+            evaluatedRootNode.name
+          }" due to an expected bail-out: ${error.message}`;
```

Comment:
Would the "expected bail-out" language reach users of the React compiler? If so, they may not know what to make of it.